### ISSUE TITLE
Installing with RPATH overwrite fails in libcxx when libc++.so.1.0 is a linker script (LIBCXX_ENABLE_ABI_LINKER_SCRIPT is ON)

### ISSUE THREAD

**Issue by arcivanov** [llvm/llvm-project#61738]
```console
FAILED: libcxx/src/CMakeFiles/install-cxx /home/tcwg-buildbot/worker/linaro-armv8-libcxx-03/llvm-project/libcxx-ci/build/armv8/libcxx/src/CMakeFiles/install-cxx
cd /home/tcwg-buildbot/worker/linaro-armv8-libcxx-03/llvm-project/libcxx-ci/build/armv8/libcxx/src && /usr/local/bin/cmake -DCMAKE_INSTALL_COMPONENT=cxx -P /home/tcwg-buildbot/worker/linaro-armv8-libcxx-03/llvm-project/libcxx-ci/build/armv8/libcxx/cmake_install.cmake
-- Install configuration: "RelWithDebInfo"
-- Installing: /home/tcwg-buildbot/worker/linaro-armv8-libcxx-03/llvm-project/libcxx-ci/build/armv8/install/lib/libc++.so.1.0
-- Installing: /home/tcwg-buildbot/worker/linaro-armv8-libcxx-03/llvm-project/libcxx-ci/build/armv8/install/lib/libc++.so.1
-- Set runtime path of "/home/tcwg-buildbot/worker/linaro-armv8-libcxx-03/llvm-project/libcxx-ci/build/armv8/install/lib/libc++.so.1.0" to "$ORIGIN"
-- Installing: /home/tcwg-buildbot/worker/linaro-armv8-libcxx-03/llvm-project/libcxx-ci/build/armv8/install/lib/libc++.so
CMake Error at cmake_install.cmake:88 (file):
  file RPATH_CHANGE could not write new RPATH:
 
    $ORIGIN
 
  to the file:
 
    /home/tcwg-buildbot/worker/linaro-armv8-libcxx-03/llvm-project/libcxx-ci/build/armv8/install/lib/libc++.so
 
  The file format is not recognized.
Call Stack (most recent call first):
  /home/tcwg-buildbot/worker/linaro-armv8-libcxx-03/llvm-project/libcxx-ci/build/armv8/libcxx/cmake_install.cmake:48 (include)
```

**Comment by arcivanov** [comment]
This is uncovered as part of testing the patch for #54955

The full build logs are here: https://buildkite.com/llvm-project/libcxx-ci/builds/21399#01872108-a209-4092-a5eb-6716ad1256ef

**Comment by arcivanov** [comment]
This is a CMake problem: https://gitlab.kitware.com/cmake/cmake/-/issues/24647
It should be fixed in 3.27.0.
